pokemon-checker/src/factories/PokemonFactory.ts

```diff
@@ -11,20 +16,82 @@ export class PokemonFactory {
     this.moveFactory = new MoveFactory();
   }
 
-  public createPokemon = (pokemon: IPokemonData): PokemonDTO => {
-    const opts: PokemonConstructorOptions = {
-      name: pokemon.name,
-      id: pokemon.id,
-      types: pokemon.types.map((slot) => slot.type),
-      moves: pokemon.moves.map((moveData) =>
+  private getFullPokemonConstructorProps = (
+    data: IPokemonData
+  ): PokemonConstructorOptions => {
+    return {
+      name: data.name,
+      id: data.id,
+      types: data.types.map((slot) => slot.type),
+      moves: data.moves.map((moveData) =>
         this.moveFactory.createMoveFromStub(moveData)
       ),
-      sprites: pokemon.sprites,
-      stats: pokemon.stats.map((statData) => ({
+      sprites: data.sprites,
+      stats: data.stats.map((statData) => ({
         base_stat: statData.base_stat,
       })),
     };
+  };
+
+  /**
+   * Creates pokemon data that can be used for rendering
+   * @param pokemon Full data from the individual pokemon requests
+   */
+  public createPokemon = (pokemon: IPokemonData): PokemonDTO => {
+    return new PokemonDTO(this.getFullPokemonConstructorProps(pokemon));
+  };
 
+  /**
+   * Creates a partial pokemon DTO
+   * @param pokemon Partial stub data from the summary call
+   */
+  public createPokemonStub = (pokemon: IPokemonStub): PokemonDTO => {
+    // TODO: Introduce a null-type element
+    const dummyType: ITypeData = {
```

Comment:
I defaulted to the first element type I could find. We should probably have a `NULL` entry but that's what the TODO is for.